test/banach/pagination_test.clj

```diff
@@ -0,0 +1,18 @@
+(ns banach.pagination-test
+  (:require [banach.pagination :as p]
+            [clojure.test :refer [deftest is]]
+            [manifold.deferred :as md]
+            [manifold.stream :as ms]))
+
+(deftest paginated->stream-test
+  (let [gets (atom 0)
+        get! (fn [[marker n]]
+               (is (= ::todo marker))
+               (swap! gets inc)
+               (md/success-deferred
+                {:results (map (partial + (* 10 n)) (range 10))
+                 :todo (when (< n 9) [::todo (inc n)])}))
+        s (p/paginated->stream get! [::todo 0] :todo :results)]
+    (is (< 0 @gets 10))
```

Comment:
And on re-reading this, I think I understand - start the stream paging process, make sure it has begun performing some reads between 0 and the max. Then, convert the results of the stream to a seq, then make sure that all gets have been  performed.